services/net/transcription/TranscriptionManager.cs

```diff
@@ -0,0 +1,252 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.CognitiveServices.Speech;
+using Microsoft.CognitiveServices.Speech.Audio;
+using TNO.Services.Managers;
+using TNO.Services.Transcription.Config;
+using TNO.API.Areas.Services.Models.Content;
+using TNO.Models.Kafka;
+using Confluent.Kafka;
+using TNO.API.Areas.Editor.Models.Lookup;
+using System.Text.Json;
+using System.Text;
+using TNO.Models.Extensions;
+
+namespace TNO.Services.Transcription;
+
+/// <summary>
+/// TranscriptionManager class, provides a Kafka Consumer service which imports audio from all active topics.
+/// </summary>
+public class TranscriptionManager : ServiceManager<TranscriptionOptions>
+{
+    #region Variables
+    #endregion
+
+    #region Properties
+    /// <summary>
+    /// get - Kafka Consumer object.
+    /// </summary>
+    protected IKafkaListener Consumer { get; private set; }
+
+    /// <summary>
+    /// get - Lookup values from the API.
+    /// </summary>
+    public LookupModel? Lookups { get; private set; }
+
+    #endregion
+
+    #region Constructors
+    /// <summary>
+    /// Creates a new instance of a TranscriptionManager object, initializes with specified parameters.
+    /// </summary>
+    /// <param name="kafka"></param>
+    /// <param name="api"></param>
+    /// <param name="options"></param>
+    /// <param name="logger"></param>
+    public TranscriptionManager(
+        IKafkaListener kafka,
+        IApiService api,
+        IOptions<TranscriptionOptions> options,
+        ILogger<TranscriptionManager> logger)
+        : base(api, options, logger)
+    {
+        this.Consumer = kafka;
+    }
+    #endregion
+
+    #region Methods
+    /// <summary>
+    /// Listen to active topics and import content.
+    /// </summary>
+    /// <returns></returns>
+    public override async Task RunAsync()
+    {
+        var delay = _options.DefaultDelayMS;
+
+        // Always keep looping until an unexpected failure occurs.
+        while (true)
+        {
+            if (this.State.Status != ServiceStatus.Running)
+            {
+                this.Logger.LogDebug("The service is not running '{Status}'", this.State.Status);
+            }
+            else
+            {
+                try
+                {
+                    // TODO: Handle e-tag.
+                    // TODO: Update on some kind of interval.
+                    this.Lookups = await _api.GetLookupsAsync();
+
+                    // Listen to every enabled data source with a topic.
+                    var topics = !String.IsNullOrWhiteSpace(_options.Topics) ? _options.GetTopics() : this.Lookups?.DataSources
+                        .Where(ds => ds.IsEnabled &&
```

Comment:
I don't think this service will look at Data Sources to determine which topics to listen to.  The Data Source is setup for ingest.  This Transcription Service isn't ingesting, it's updating existing content we've ingested.

Reply:
Fixed.